### PR TITLE
Add Sapling note locking to the CWallet class

### LINKED ISSUE #3442: Add Sapling Note locking and unlocking to CWallet
Override methods to take SaplingOutPoint etc.
```
void CWallet::LockNote(const JSOutPoint& output)
void CWallet::UnlockNote(const JSOutPoint& output)
void CWallet::UnlockAllNotes()
bool CWallet::IsLockedNote(const JSOutPoint& outpt) const
std::vector<JSOutPoint> CWallet::ListLockedNotes()
```

### PR DESCRIPTION
**Pull request by bitcartel**
Closes #3442.

### CODE REVIEW

File: src/wallet/wallet.h [hunk 1]
```diff
@@ -941,7 +941,7 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     CPubKey vchDefaultKey;
 
     std::set<COutPoint> setLockedCoins;
-    std::set<JSOutPoint> setLockedNotes;
+    std::set<JSOutPoint> setLockedSproutNotes;
```
Comment (str4d) [inline src/wallet/wallet.h]: The commit message incorrectly says this is a rename to `setLockedSaplingNotes`.

**Comment by bitcartel** [comment]
Force-pushed to (1) fix commit message (2) rebase on master

**Comment by str4d** [comment]
re-ut(ACK+cov)

**Comment by str4d** [comment]
Correct. A subsequent PR should address that 🙂

@zkbot r+

**Comment by bitcartel** [comment]
"Error: Error: Disk space is low!" failure in debian8-kitchensink, and transient failures on macos, centos, arch

@zkbot retry

**Pull request merged**